src/lib.rs

```diff
@@ -178,7 +174,6 @@ impl TestSuite {
     /// Create a new `TestSuite` with a given name
     pub fn new(name: &str) -> TestSuite {
```

Comment:
if we add something like `with_id` or make `new` take an `id` that would be nice.

Reply:
The issue I see with ID is that it is part of the `TestSuite` but it's value is usually determined by the containing the `TestSuite`. I'm not entirely sure what's the best way to handle this.